phoenix-core-server/src/main/java/org/apache/phoenix/replication/log/LogFile.java

```diff
@@ -0,0 +1,579 @@
+        }
+
+       /**
+         * Builds an HBase mutation, given a Record.
+         * @param mutation The Record instance.
+         * @return The HBase Mutation
+         */
+        static Mutation toHBaseMutation(Record record) {
+            if (record == null) {
+                return null;
+            }
+            return LogFileRecord.toHBaseMutation(record);
+        }
+    }
+
+    /**
+     * Defines the types of mutations that can be stored in a Record.
+     * Mirrors relevant parts of HBase Cell.Type.
+     */
+    enum MutationType {
+        PUT((byte) 4),
+        DELETE((byte) 8),                 // Delete for entire row at specified ts
+        DELETE_COLUMN((byte) 12),         // Delete for specific column at specified ts
+        DELETE_FAMILY((byte) 14),         // Delete for entire column family at specified ts
+        DELETE_FAMILY_VERSION((byte) 10); // Delete for specific column family version
+
+        private final byte code;
+
+        MutationType(byte code) {
+            this.code = code;
+        }
+
+        public byte getCode() {
+            return code;
+        }
+
+        public static MutationType codeToType(byte code) {
+            for (MutationType t : values()) {
+                if (t.code == code) {
+                    return t;
+                }
+            }
+            throw new IllegalArgumentException("Unknown MutationType code: " + code);
+        }
+
+        public static MutationType fromHBaseCellType(Cell.Type type) {
+            switch (type) {
+                case Put:
+                    return PUT;
+                case Delete:
+                    return DELETE;
+                case DeleteColumn:
+                    return DELETE_COLUMN;
+                case DeleteFamily:
+                    return DELETE_FAMILY;
+                case DeleteFamilyVersion:
+                    return DELETE_FAMILY_VERSION;
+                default:
+                    throw new IllegalArgumentException("Unsupported MutationType: " + type);
+            }
+        }
+    }
+
+    /** Interface for writing replication logs */
+    interface Writer extends Closeable {
+        /**
+         * Initializes the writer with necessary context.
+         * @param context The context containing initialization parameters.
+         * @throws IOException if initialization fails.
+         */
+        void init(LogFileWriterContext context) throws IOException;
+
+        /**
+         * Appends an HBase mutation to the log file. The log record may be buffered internally.
+         * @param schemaObjectName The schema object name
+         * @param commitId The commit identifier
+         * @param mutation The mutation to append.
+         * @throws IOException if an I/O error occurs during append.
+         */
+        void append(String schemaObjectName, long commitId, Mutation mutation) throws IOException;
```

Comment:
schemaObjectName should be renamed as hbaseTableName or physicalTableName. This object is not a Phoenix object.